testing/test_terminal.py

```diff
@@ -1598,23 +1619,23 @@ def test_failure():
 
 
 @pytest.mark.filterwarnings("default")
-def test_terminal_no_summary_warnings_header_once(testdir):
-    testdir.makepyfile(
+def test_terminal_no_summary_warnings_header_once(pytester: Pytester) -> None:
+    pytester.makepyfile(
         """
         def test_failure():
             import warnings
             warnings.warn("warning_from_" + "test")
             assert 0
     """
     )
-    result = testdir.runpytest("--no-summary")
+    result = pytester.runpytest("--no-summary")
     result.stdout.no_fnmatch_line("*= warnings summary =*")
     result.stdout.no_fnmatch_line("*= short test summary info =*")
 
 
-@pytest.fixture(scope="session")
-def tr() -> TerminalReporter:
-    config = _pytest.config._prepareconfig()
+@pytest.fixture(scope="function")
+def tr(pytester: Pytester) -> TerminalReporter:
+    config = pytester.parseconfig()
```

Comment:
I believe with the chdir fix above this can remain session-scoped:

```suggestion
@pytest.fixture(scope="session")
def tr() -> TerminalReporter:
    config = _pytest.config._prepareconfig()
```

Reply:
reverted in 009e04c